src/main/java/hudson/matrix/MatrixBuild.java

```diff
@@ -91,34 +91,32 @@ public Object readResolve() {
     }
 
     /**
+     * Delete build and its child builds.
+     *
+     * Child builds can be skipped if they have to be kept.
+     *
+     * @since TODO
+     */
+    @Override
+    public void delete() throws IOException {
+        super.delete();
+
+        for(MatrixRun run : getExactRuns()) {
+            if (run.getWhyKeepLog() != null) continue;
```

Comment:
OK, so long as you have tested that the behavior makes some kind of sense when the child is kept while the parent is deleted. I guess the child is still accessible from the `MatrixConfiguration`.